__tests__/app.test.js

```diff
@@ -313,5 +312,78 @@ describe("Using app.js to run the database of NC-games", () => {
+      return request(app)
+        .get("/api/reviews")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.reviewArray.length > 0).toBe(true);
+          expect(
+            body.reviewArray.forEach((category) => {
+              expect(category).toHaveProperty("owner", expect.any(String));
+              expect(category).toHaveProperty("title", expect.any(String));
+              expect(category).toHaveProperty("review_id", expect.any(Number));
+              expect(category).toHaveProperty("category", expect.any(String));
+              expect(category).toHaveProperty(
+                "review_img_url",
+                expect.any(String)
+              );
+              expect(category).toHaveProperty("created_at", expect.any(String));
+              expect(category).toHaveProperty("votes", expect.any(Number));
+              expect(category).toHaveProperty("designer", expect.any(String));
+              expect(category).toHaveProperty(
+                "comment_count",
+                expect.any(String)
+              );
+            })
+          );
+        });
+    });
+    test("200: response with a Array of objects with reviews but category is dexterity", () => {
+      return request(app)
+        .get("/api/reviews?category=dexterity")
+        .expect(200)
+        .then(({ body }) => {
+          expect(Array.isArray(body.reviewArray)).toBe(true);
+          expect(body.reviewArray.length > 0).toBe(true);
+          expect(
+            body.reviewArray.forEach((category) => {
+              expect(category).toHaveProperty("owner", expect.any(String));
+              expect(category).toHaveProperty("title", expect.any(String));
+              expect(category).toHaveProperty("review_id", expect.any(Number));
+              expect(category).toHaveProperty("category", "dexterity");
+              expect(category).toHaveProperty(
+                "review_img_url",
+                expect.any(String)
+              );
+              expect(category).toHaveProperty("created_at", expect.any(String));
+              expect(category).toHaveProperty("votes", expect.any(Number));
+              expect(category).toHaveProperty("designer", expect.any(String));
+              expect(category).toHaveProperty(
+                "comment_count",
+                expect.any(String)
+              );
+            })
+          );
+        });
+    });
+    test("404: Not Found from possible spelling mistake", () => {
+      return request(app)
+        .get("/api/reveiws")
+        .expect(404)
+        .then((response) => {
+          expect(response.body).toEqual({ msg: "Not Found" });
+        });
+    });
+    test("404: response with a NOT FOUND as category is apples and is not there", () => {
+      return request(app)
+        .get("/api/reviews?category=apples")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body).toEqual({ status: 404, msg: "Not Found" });
+        });
+    });
+  });
```

Comment:
How about a test where we give an incorrect query key like 'topic=dexterity'

Reply:
Done